services/delivery/src/deliver/events.js

```diff
@@ -30,12 +31,14 @@ module.exports = {
   async click(adunit, correlator, adId, { now, email, send }) {
     const { _id: adunitId, deploymentId, publisherId } = adunit;
 
-    const view = await db.findOne('views', { adId, correlator }, { projection: { _id: 1 } });
+    const type = 'view';
+    const view = await db.findOne('events', { type, adId, correlator }, { projection: { _id: 1 } });
```

Comment:
This query should be indexed...

Reply:
You also don't need to specify `type` as a const... just include it in the findOne query directly